### PR TITLE
MULE-13057: Cannot access attachments without Content-Disposition name attribute

### PR DESCRIPTION
**Pull request by afelisatti**

### CODE REVIEW

**Comment by afelisatti** [comment]
REVIEW ONLY: This is pending a full test.

File: modules/http/src/main/java/org/mule/module/http/internal/multipart/HttpPartDataSource.java [hunk 1]
```diff
@@ -92,9 +91,20 @@ public String getName()
...
         final Map<String, DataHandler> httpParts = new HashMap<>(parts.size());
+        int anonymousPartCount = 0;
         for (HttpPart part : parts)
         {
-            httpParts.put(part.getName(), new DataHandler(new HttpPartDataSource(part)));
+            String name;
+            if (part.getName() == null)
+            {
+                name = ANONYMOUS_ATTACHMENT_PREFIX + valueOf(anonymousPartCount);
+                anonymousPartCount++;
```
Comment (elrodro83) [inline modules/http/src/main/java/org/mule/module/http/internal/multipart/HttpPartDataSource.java]: you can do this increase in the usage in the previous line

File: modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java [hunk 2]
```diff
@@ -210,6 +210,43 @@ public void returnOnlyOneContentTypeHeaderPerPart() throws Exception
...
+            final MuleMessage receivedMessage = muleContext.getClient().request(VM_MESSAGE_ENDPOINT, 1000);
+            assertThat(receivedMessage.getInboundAttachmentNames().size(), is(1));
+            HttpPartDataSource muleAttachment = (HttpPartDataSource) receivedMessage.getInboundAttachment("mule_attachment_0").getDataSource();
+            assertThat(muleAttachment.getHeader("Custom-Header"), equalTo("custom"));
+            assertThat(muleAttachment.getContentType(), equalTo("application/pdf"));
+            assertThat(muleAttachment.getHeader(CONTENT_TRANSFER_ENCODING), equalTo("base64"));
+
+            // check that the HTTP response includes the same headers as the HTTP request (when echoing)
+            final Collection<HttpPart> parts = HttpParser.parseMultipartContent(response.getResponseBodyAsStream(), response.getContentType());
+            assertThat(parts.size(), Is.is(1));
```
Comment (elrodro83) [inline modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java]: static import

File: modules/http/src/main/java/org/mule/module/http/internal/multipart/HttpPartDataSource.java [hunk 3]
```diff
@@ -26,7 +26,7 @@
 
 public class HttpPartDataSource implements DataSource
 {
-    public static final String ANONYMOUS_ATTACHMENT_PREFIX = "mule_attachment_";
+    public static final String ANONYMOUS_ATTACHMENT_FORMAT = "mule_attachment_%s";
```
Comment (elrodro83) [inline modules/http/src/main/java/org/mule/module/http/internal/multipart/HttpPartDataSource.java]: %d

File: modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java [hunk 4]
```diff
@@ -238,7 +238,7 @@ public void receiveMultipartMixed() throws Exception {
             assertThat(muleAttachment.getHeader(CONTENT_TRANSFER_ENCODING), equalTo("base64"));
 
             // check that the HTTP response includes the same headers as the HTTP request (when echoing)
-            final Collection<HttpPart> parts = HttpParser.parseMultipartContent(response.getResponseBodyAsStream(), response.getContentType());
+            final Collection<HttpPart> parts = parseMultipartContent(response.getResponseBodyAsStream(), response.getContentType());
             assertThat(parts.size(), Is.is(1));
```
Comment (elrodro83) [inline modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java]: static import

File: modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java [hunk 5]
```diff
@@ -279,7 +279,7 @@ private void processAttachmentRequestAndResponse(String pathToCall, String expec
                 final String contentType = response.getFirstHeader(HttpHeaders.Names.CONTENT_TYPE).getValue();
                 assertThat(contentType, Matchers.containsString(expectedResponseContentType));
 
-                final Collection<HttpPart> parts = HttpParser.parseMultipartContent(response.getEntity().getContent(), contentType);
+                final Collection<HttpPart> parts = parseMultipartContent(response.getEntity().getContent(), contentType);
                 assertThat(parts.size(), Is.is(2));
```
Comment (elrodro83) [inline modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java]: static import

File: modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java [hunk 6]
```diff
@@ -154,7 +154,7 @@ public void receiveAttachmentWithCustomHeaders() throws Exception
             assertThat(muleAttachment.getHeader(CONTENT_DISPOSITION), equalTo("file; name=\"partName\"; documentId=1"));
 
             // check that the HTTP response includes the same headers as the HTTP request (when echoing)
-            final Collection<HttpPart> parts = HttpParser.parseMultipartContent(response.getResponseBodyAsStream(), response.getContentType());
+            final Collection<HttpPart> parts = parseMultipartContent(response.getResponseBodyAsStream(), response.getContentType());
             assertThat(parts.size(), Is.is(1));
```
Comment (elrodro83) [inline modules/http/src/test/java/org/mule/module/http/functional/listener/HttpListenerAttachmentsTestCase.java]: syatyic import

**Comment by afelisatti** [comment]
This has been validated and can now be merged.

**Pull request merged**